google/data_source_google_folder.go

```diff
@@ -0,0 +1,59 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	resourceManagerV2Beta1 "google.golang.org/api/cloudresourcemanager/v2beta1"
+	"google.golang.org/api/googleapi"
+)
+
+func dataSourceGoogleFolder() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleFolderRead,
+
+		Schema: map[string]*schema.Schema{
+			"parent": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"display_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceGoogleFolderRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	parent := d.Get("parent").(string)
+	displayName := d.Get("display_name").(string)
+
+	queryString := fmt.Sprintf("lifecycleState=ACTIVE AND parent=%s AND displayName=%s", parent, displayName)
```

Comment:
could you rename the data source to `google_active_folder` to be explicit about the fact that you can only fetch active folder.

Reply:
Will do, that makes a lot of sense.